HAL/keymaster/4.1/JavacardKeymaster4Device.cpp

```diff
@@ -990,16 +1200,34 @@ Return<void> JavacardKeymaster4Device::finish(uint64_t operationHandle, const hi
                 return errorCode;
             }
 
+            //In case if there is ASSOCIATED_DATA present in the keyparams, then make sure it is either passed with
+            //update call or finish call. Don't send ASSOCIATED_DATA in both update and finish calls. aadTag is used to
+            //check if ASSOCIATED_DATA is already sent in update call. If addTag is true then skip ASSOCIATED_DATA from
+            //keyparams.
```

Comment:
Add finish call at te end in the comment.
If addTag is true then skip ASSOCIATED_DATA from keyparams in finish call.

Reply:
Done.